micrometer/src/main/java/feign/micrometer/MicrometerCapability.java

```diff
@@ -24,18 +25,35 @@
 import io.micrometer.core.instrument.Metrics;
 import io.micrometer.core.instrument.simple.SimpleConfig;
 import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import io.micrometer.observation.ObservationRegistry;
 
 public class MicrometerCapability implements Capability {
 
   private final MeterRegistry meterRegistry;
 
+  private final ObservationRegistry observationRegistry;
+
   public MicrometerCapability() {
-    this(new SimpleMeterRegistry(SimpleConfig.DEFAULT, Clock.SYSTEM));
+    this(new SimpleMeterRegistry(SimpleConfig.DEFAULT, Clock.SYSTEM), ObservationRegistry.NOOP);
     Metrics.addRegistry(meterRegistry);
   }
 
   public MicrometerCapability(MeterRegistry meterRegistry) {
+    this(meterRegistry, ObservationRegistry.NOOP);
+  }
+
+  public MicrometerCapability(MeterRegistry meterRegistry,
+      ObservationRegistry observationRegistry) {
     this.meterRegistry = meterRegistry;
+    this.observationRegistry = observationRegistry;
+  }
+
+  @Override
+  public <B extends BaseBuilder<B>> BaseBuilder<B> enrich(BaseBuilder<B> baseBuilder) {
+    if (!observationRegistry.isNoop()) {
+      baseBuilder.clientInterceptor(new ObservedClientInterceptor(observationRegistry));
```

Comment:
no good. use `enrich(ClientInterceptor)` instead.

Reply:
That doesn't make sense cause I don't want to enrich an existing `ClientInterceptor` - I want to want to add a ClientInterceptor whenever this capability is being used. Regardless, I've applied the suggested change and needed to change the tests accordingly (I needed to ensure that there is a `ClientInterceptor.DEFAULT` that is being enriched.

Reply:
I'm reverting that change cause it makes no sense. We would be overriding the provided client interceptor. In this case I don't think that capabilities are the way to go. Users will need to manually add a `ClientInterceptor` on a `Feign.Builder` and that's it.